source/extensions/filters/network/kafka/kafka_request.h

```diff
@@ -0,0 +1,1316 @@
+#pragma once
+
+#include <sstream>
+
+#include "envoy/common/exception.h"
+
+#include "common/common/assert.h"
+
+#include "extensions/filters/network/kafka/kafka_protocol.h"
+#include "extensions/filters/network/kafka/parser.h"
+#include "extensions/filters/network/kafka/serialization.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace Kafka {
+
+// === VECTOR ==================================================================
+
+template <typename T> std::ostream& operator<<(std::ostream& os, const std::vector<T>& arg) {
+  os << "[";
+  for (auto iter = arg.begin(); iter != arg.end(); iter++) {
+    if (iter != arg.begin()) {
+      os << ", ";
+    }
+    os << *iter;
+  }
+  os << "]";
+  return os;
+}
+
+template <typename T> std::ostream& operator<<(std::ostream& os, const absl::optional<T>& arg) {
+  if (arg.has_value()) {
+    os << *arg;
+  } else {
+    os << "<null>";
+  }
+  return os;
+}
+
+// === REQUEST HEADER ==========================================================
+
+struct RequestHeader {
```

Comment:
is it really called request in Kafka? isn't it more of a message/pub/sub or operation?

Reply:
the underlying entities are really called requests/responses
see https://kafka.apache.org/protocol#protocol_common & https://kafka.apache.org/protocol#protocol_messages (and corresponding classes in java client like `ProduceRequest`)